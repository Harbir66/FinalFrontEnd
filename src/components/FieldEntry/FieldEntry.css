.field-entry{
    display:flex;
    width: 49.9vw;
    justify-content: space-between;
    background-color: white;
    box-shadow: 0 0 1px 1px  #D4D6E5;
    border-radius: 1vh;
    height: 6vh;
}

.first-portion{
    display: flex;
    width: 90%;
    align-items: center;
    gap: 1vh 0;
}

.tab{
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4vw;
    height: 6vh;
    background-color: #7691FF;
    border-radius: 1vh 0 0 1vh;
    color: white;
    font-size: 2vh;
}

.field-name{
    display: flex;
    align-items: center;
    justify-content: flex-start;
    padding-left: 1vw;
    width: 20vw;
    height: 6vh;
    background-color: #F4F6FF;
    color: #4F4F4F;
    font-size: 2vh;
}

.field-type{
    display: flex;
    align-items: center;
    justify-content: flex-start;
    width: 20vw;
    height: 6vh;
    background-color: #F4F6FF;
    color: #5c5c5c;
    font-size: 1.5vh;
}

.field-buttons{
    display: flex;
    align-items: center;
    width: 10vw;
    justify-content: flex-start;
    background-color: #F4F6FF;
    border-radius: 0 1vh 1vh 0;
}

.field-buttons > button{
    border: none;
    background: none;
    color: #5c5c5c;
}

.field-buttons > button > img{
    height: 2vh;
}