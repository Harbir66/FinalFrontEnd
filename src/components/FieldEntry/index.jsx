import React from 'react';
import './FieldEntry.css';
import PropTypes from 'prop-types';
import editIcon from '../../assets/user-edit-text-message-note@3x.png';
import deleteIcon from '../../assets/trash-delete-recycle-bin-bucket-waste@3x.png';
import Modal from '../Modal';

function FieldEntry({ fieldName, handleFieldDelete, handleFieldRename }) {
  const [input, setInput] = React.useState('');
  const [isOpen, setIsOpen] = React.useState(false);
  const openModal = () => setIsOpen(true);
  const closeModal = () => setIsOpen(false);
  const handleSubmit = (e) => {
    e.preventDefault();
    handleFieldRename(fieldName, input);
    setInput('');
    closeModal();
  };
  const modal = (
    <Modal isOpen={isOpen} onClose={closeModal}>
      <h1>Rename the Field</h1>
      <form>
        <label htmlFor="name">
          New name of the Field
          <input
            type="text"
            id="name"
            value={input}
            onChange={(e) => setInput(e.target.value)}
          />
        </label>
        <button type="button" onClick={closeModal}>
          Cancel
        </button>
        <button onClick={handleSubmit} type="submit">
          Create
        </button>
      </form>
    </Modal>
  );
  return (
    <div className="field-entry">
      <div className="first-portion">
        <span className="tab">Ab</span>
        <span className="field-name">{fieldName}</span>
        <span className="field-type">Text</span>
      </div>
      <div className="field-buttons">
        <button onClick={openModal} type="button">
          <img src={editIcon} alt="edit-icon" />
        </button>
        <button onClick={() => handleFieldDelete(fieldName)} type="button">
          <img src={deleteIcon} alt="delete-icon" />
        </button>
      </div>
      {modal}
    </div>
  );
}

export default FieldEntry;

FieldEntry.propTypes = {
  fieldName: PropTypes.string.isRequired,
  handleFieldDelete: PropTypes.func.isRequired,
  handleFieldRename: PropTypes.func.isRequired,
};
