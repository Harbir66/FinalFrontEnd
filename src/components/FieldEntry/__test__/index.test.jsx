import React from 'react';
import { render } from '@testing-library/react';
import FieldEntry from '..';

describe('FieldEntry', () => {
  it('should render', () => {
    const { asFragment } = render(
      <FieldEntry
        fieldName="test"
        handleFieldDelete={jest.fn()}
        handleFieldRename={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
