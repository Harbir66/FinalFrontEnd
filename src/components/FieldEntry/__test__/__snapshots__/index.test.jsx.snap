// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldEntry should render 1`] = `
<DocumentFragment>
  <div
    class="field-entry"
  >
    <div
      class="first-portion"
    >
      <span
        class="tab"
      >
        Ab
      </span>
      <span
        class="field-name"
      >
        test
      </span>
      <span
        class="field-type"
      >
        Text
      </span>
    </div>
    <div
      class="field-buttons"
    >
      <button
        type="button"
      >
        <img
          alt="edit-icon"
          src="user-edit-text-message-note@3x.png"
        />
      </button>
      <button
        type="button"
      >
        <img
          alt="delete-icon"
          src="trash-delete-recycle-bin-bucket-waste@3x.png"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
